pkg/apis/serving/v1alpha2/kfservice_types.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2019 kubeflow.org.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"github.com/knative/pkg/apis"
+	duckv1beta1 "github.com/knative/pkg/apis/duck/v1beta1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KFServiceSpec defines the desired state of KFService
+type KFServiceSpec struct {
+	// Default defines the configuration of KFService
+	Default ServicesSpec `json:"default"`
+	// Canary defines an alternate KFService configuration to route a percentage of traffic.
+	Canary *ServicesSpec `json:"canary,omitempty"`
+	// Traffic split percentage between default and canary KFService
+	CanaryTrafficPercent int `json:"canaryTrafficPercent,omitempty"`
+}
+
+type ServicesSpec struct {
+	// Optional prepost processing specification
+	Prepostprocess *PrepostprocessSpec `json:"prepostprocess,omitempty"`
+	// Predictor for a given model
+	Predict ModelSpec `json:"predict"`
+	// Optional Explain specification to add a model explainer next to the chosen predictor.
+	Explain *ExplainSpec `json:"explain,omitempty"`
+	// Default deployment specification
+	Deployment *DeploymentSpec `json:"deployment,omitempty"`
+	// Service account name
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+}
+
+type DeploymentSpec struct {
```

Comment:
Cool. I think this is a great abstraction. Should it also include ServiceAccountName?